src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -156,9 +187,41 @@ const ChatItemMyTextV2 = ({
             <Text style={styles.timeText}>{time}</Text>
           </View>
         )}
-        <Text ref={messageRef} style={styles.text}>
-          {`${message}`}
-        </Text>
+        {attachments.length > 0 && (
+          <View style={styles.attachmentContainer}>
+            {attachments
+              .filter((item, index) => index <= 3)
+              .map((item, index) => (
+                <View
+                  key={item.thumb_url}
+                  style={{
+                    width: `${
+                      (attachments.length >= 3 && index > 0) || attachments.length >= 4 ? 50 : 100
+                    }%`,
+                    height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
+                    position: 'relative',
+                    overflow: 'hidden'
+                  }}>
+                  <FastImage
+                    style={styles.image}
+                    source={{
+                      uri: item.thumb_url
+                    }}
+                  />
+                  {attachments.length > 4 && index === 3 && (
+                    <View style={styles.moreOverlay}>
+                      <Text style={styles.moreText}>+{attachments.length - 4}</Text>
+                    </View>
+                  )}
+                </View>
+              ))}
+          </View>
+        )}
+        {message?.trim() !== '' && (
+          <Text ref={messageRef} style={styles.text}>
+            {`${message}`}
+          </Text>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new code block handles the rendering of attachments in the chat. It includes logic for displaying up to 4 images and an overlay for additional images. Ensure that the logic works as expected and handles edge cases (e.g., when there are no attachments, only one attachment, more than 4 attachments). Also, consider extracting this logic into a separate function or component for better readability and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->